output/cloud/expv2/output.go

```diff
@@ -122,6 +126,37 @@ func (o *Output) StopWithTestError(testErr error) error {
 	return nil
 }
 
+func (o *Output) runFlushWorkers() {
+	t := time.NewTicker(o.config.MetricPushInterval.TimeDuration())
+
+	// TODO: drop it when the migration from v1 is completed
+	workers := 4
+	if o.config.MetricPushConcurrency.Int64 > 1 &&
+		o.config.MetricPushConcurrency.Int64 < 10 {
+		workers = int(o.config.MetricPushConcurrency.Int64)
+	}
+	for i := 0; i < workers; i++ {
+		o.wg.Add(1)
+		go func() {
+			defer func() {
+				t.Stop()
+				o.wg.Done()
+			}()
+
+			for {
+				select {
+				case <-t.C:
+					o.flushMetrics()
```

Comment:
Because of how `flush` is implemented. This in most cases will lead to *one* worker doing the work either way.

`flush` gets *all* the metrics each time it is called and then splits them if necessary.
So with multiple workers they will all try to get *all* the metric series and only *one* will succeed.

If that worker manages to flush them in the Push Interval - it will just come back to wait for this ticker.

This will still help if flush takes longer then the push interval *IF* the actual context used for that is actually not the pushInterval - so in practice this will never use more then 1 worker.

I am very much for:
1. increasing the context and timeouts for this push *significantly*. There actually is a cloudAPI default timeout which is 1minute. 
2. The timeout must be reset every time we retry. Currently, it doesn't so in practice if it times outs it will timeout again and again. 
3. Preferably something will every push interval trigger, get all the time series, split them if necessary *and* then give it to the workers so they can "only" do the marshalling, encoding and sending.

(3) can likely be skipped as the first 2 will likely help with that a lot.

https://github.com/grafana/k6/blob/9254ce7ddafe856ea58dea46f54049a8248fcfe1/output/cloud/expv2/output.go#L177C1-L191 
for the flushMetric code and showcasing where it has a context with the timeout

https://github.com/grafana/k6/blob/9254ce7ddafe856ea58dea46f54049a8248fcfe1/output/cloud/expv2/flush.go#L26-L31 

the `flush` code that drains the whole buckets


As an additional note, because we are using the same `http.Client` underneath if this is actually `http2` there are chances that we will be hitting the same ingest server over and over again, which might not be what we want.

But this will likely need more investigation and is a problem even for cloud output v1, so probably a thing we can do much later

Reply:
Part of https://github.com/grafana/k6/issues/3117 

Under > "reevaluate the current periodic and abort signal architecture/interaction"